2230/3/rap-battle/rap-battle.rb

```diff
@@ -0,0 +1,131 @@
+require 'trollop' # Trollop is a commandline option parser for Ruby
+require 'russian_obscenity' # Gem for filtering russian obscene language
+
+opts = Trollop.options do
+  opt %s[top-bad-words], 'Prints <i> records from the Top Bad Words', default: 0
+  # opt %s[top-words], 'Prints <i> words per people or list rappers if run without --name', :default => 30
+  # opt :name, 'Name of the rapper. Use with --top-words', :type => :string
+end
+
+# rules for ARGV parametrs
+# Trollop::educate if ARGV.empty?
+Trollop.die %s[top-bad-words], 'must be non-negative' if opts[%s[top-bad-words]] <= 0
+# p opts
+
+# Generate list of files
+# rubocop:disable Style/InverseMethods
+@common_file_list = Dir.entries('data').select { |file_name| !File.directory? file_name }
```

Comment:
instance variables without classes? not bad 🤣 
Refactor this file in OOP style also, do not remember about OOP and SOLID principles, including single responsibility principle.